### ISSUE TITLE
HiGHS for column generation

### ISSUE THREAD

**Issue by andrea-cassioli-maersk** [ERGO-Code/HiGHS#1465]
Hi,
I am considering HiGHS for to implement a column generation approach, but it is unclear to me how I should proceed.

For what I can see the way is to 

* create the master problem as a HighsModel
* solve the master and so on (pricer...)
* add new columns using addColumns (is it the efficient way?)
* solve the master again using warm start

the last point is what troubles me the most: 

* how does HiGHS knows the model has changed?
* how do I make sure HiGHS warm starts? 
* Shall I save the basis and last solution and feed it back?

**Comment by jajhall** [comment]
That makes about four projects that I know of using HiGHS for column generation!

What you described is correct and will be efficient. If you add a column using 'Highs::addCol' then HiGHS will automatically update its internal basis - the new column will be nonbasic - and won't even have to refactor the basis matrix when you next call 'Highs::run'. 

You're talking as if you'll be solving the master with simplex. There is a strong argument to use interior point, since that yields a possibly valuable stabilisation of the dual values. The downside is that interior point doesn't hot start.

**Comment by andrea-cassioli-maersk** [comment]
@jajhall yeah, we focus on primal simplex, but we could experiment with IP as well.

Is 'Highs::addColumns' part of the model or the solver? I cannot really find a complete documentation of the various interfaces...
